lib/ansible/modules/user.py

```diff
@@ -3204,7 +3242,10 @@ def main():
         result['home'] = info[5]
         result['shell'] = info[6]
         if user.groups is not None:
-            result['groups'] = user.groups
+            result['groups'] = ','.join(sorted(user.groups_diff['after']))
```

Comment:
Should these be strings? I'd expect them to be like JSON lists in the module output for easier post-processing of the results..

Reply:
That's the type listed in the return docs too. The input is also a comma-separated string.

Reply:
Oh, it's not the input for `groups`. So the return type was maybe just wrong.

Reply:
verified that the return type is indeed a comma-separated string still on devel: https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/user.py#L584-L585

user.groups isn't set anywhere else

Reply:
Got it.